### PR TITLE
CakeTime::listTimezones(): Add option to Display timezone abbreviations

### PR DESCRIPTION
**Pull request by rchavik**
Useful for countries that do not have many of its cities, even major ones,
listed. For eg: Indonesia, only have 4 cities listed.

For backward compatibility, abbreviations will not be shown.

### CODE REVIEW

File: lib/Cake/Utility/CakeTime.php [hunk 1]
```diff
@@ -1075,12 +1075,19 @@ public static function i18nFormat($date, $format = null, $default = false, $time
...
  * @return array List of timezone identifiers
  * @since 2.2
  * @link http://book.cakephp.org/2.0/en/core-libraries/helpers/time.html#TimeHelper::listTimezones
  */
-	public static function listTimezones($filter = null, $country = null, $group = true) {
+	public static function listTimezones($filter = null, $country = null, $options = array()) {
+		$options = array_merge(array(
+			'group' => true,
+			'abbr' => false,
+		), $options);
```
Comment (markstory) [inline lib/Cake/Utility/CakeTime.php]: This isn't fully backwards compatible as sometimes $options will be a boolean. Which cannot be array merged.

**Comment by markstory** [comment]
I like the idea, just need to get the tests passing.

**Comment by rchavik** [comment]
something weird in travis.  looking into it

File: lib/Cake/Utility/CakeTime.php [hunk 2]
```diff
@@ -1075,12 +1075,29 @@ public static function i18nFormat($date, $format = null, $default = false, $time
...
+ * 	Otherwise, an array containing `group`, `abbr`, `before`, and `after` keys.
+ * 	Setting `group` and `abbr` to true will group results and append timezone
+ * 	abbreviation in the display value. Set `before` and `after` to customize
+ * 	the abbreviation wrapper.
  * @return array List of timezone identifiers
  * @since 2.2
  * @link http://book.cakephp.org/2.0/en/core-libraries/helpers/time.html#TimeHelper::listTimezones
  */
-	public static function listTimezones($filter = null, $country = null, $group = true) {
+	public static function listTimezones($filter = null, $country = null, $options = array()) {
```
Comment (ADmad) [inline lib/Cake/Utility/CakeTime.php]: Method signature should not be changed in bugfix release.

File: lib/Cake/Utility/CakeTime.php [hunk 3]
```diff
@@ -1075,12 +1075,29 @@ public static function i18nFormat($date, $format = null, $default = false, $time
...
+		if ($options === true || $options === false) {
```
Comment (ADmad) [inline lib/Cake/Utility/CakeTime.php]: `if (is_bool($options))`

**Comment by rchavik** [comment]
Now rebased on top of 2.8. Actual diff is now 19c5231896ce3c63545031595a23d9d6241b70ee

Also, it's using `is_bool()` now.

**Comment by ADmad** [comment]
The PR is still targeting 2.7 (which can't be changed on GH, need to create a new one).

**Comment by rchavik** [comment]
Fixup now on e6acacac031a2a0b

@ADmad For the love of `git`, just merge using CLI ffs :trollface:

**Comment by markstory** [comment]
Looks like this will merge 2.8 into 2.7. Lets not do that.

**Comment by rchavik** [comment]
It's already merged manually.

**Comment by markstory** [comment]
@rchavik Are you going to forward port to 3.x as well?

**Comment by rchavik** [comment]
Sure. I can do that.

**Comment by markstory** [comment]
Awesome, thanks :dancer:

**Pull request closed without merging**